source/profiles.c

```diff
@@ -149,7 +149,7 @@ bool reactiveFade(led_t *ledColors, uint8_t intensity) {
       ledColors[i].red >>= intensity;
       ledColors[i].green >>= intensity;
       ledColors[i].blue >>= intensity;
-    } else if (animatedPressedFadeBuf[i] > 0) {
```

Comment:
The underlying cause is different. `shouldUpdate` flag is used to prevent unnecessary calls to animation callback when there is nothing to light. For example, when the user has not typed anything in a while, all LEDs are disabled with this profile, and we can stop iterating until the next keypress. I am working on another refactoring and will try to create another PR today.

Reply:
Okay I will put this on hold for now then